sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveQuerySuite.scala

```diff
@@ -69,6 +69,11 @@ class HiveQuerySuite extends HiveComparisonTest with BeforeAndAfter {
     sql("DROP TEMPORARY FUNCTION udtf_count2")
   }
 
+  test("SPARK-8010: implicit promote to string type") {
+    sql("select case when true then '1' else 1 end from src ")
+    sql("select coalesce(null, 1, '1') from src ")
```

Comment:
And probably it's better to move the test into https://github.com/OopsOutOfMemory/spark/blob/pnts/sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala